hw/ip/aes/pre_syn/syn_yosys.sh

```diff
@@ -0,0 +1,152 @@
+#!/bin/bash
+
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script drives the experimental Yosys synthesis flow. More details can be found in README.md
+
+set -e
+set -o pipefail
+
+error () {
+    echo >&2 "$@"
+    exit 1
+}
+
+teelog () {
+    tee "$LR_SYNTH_OUT_DIR/log/$1.log"
+}
+
+if [ ! -f syn_setup.sh ]; then
+    error "No syn_setup.sh file: see README.md for instructions"
+fi
+
+#-------------------------------------------------------------------------
+# setup flow variables
+#-------------------------------------------------------------------------
+source syn_setup.sh
+
+#-------------------------------------------------------------------------
+# prepare output folders
+#-------------------------------------------------------------------------
+mkdir -p "$LR_SYNTH_OUT_DIR/generated"
+mkdir -p "$LR_SYNTH_OUT_DIR/log"
+mkdir -p "$LR_SYNTH_OUT_DIR/reports/timing"
+
+rm -f latest
+ln -s "$LR_SYNTH_OUT_DIR" latest
+
+#-------------------------------------------------------------------------
+# use sv2v to convert all SystemVerilog files to Verilog
+#-------------------------------------------------------------------------
+export LR_SYNTH_SRC_DIR="../../$LR_SYNTH_IP_NAME"
+
+# Get OpenTitatn dependency sources.
```

Comment:
Nit: OpenTitan.

Not for this PR, but surely we could do this by running fusesoc first? Then we'd get all the sources wrapped up nicely. That would also get rid of the sed hacks for technology mapping, I think?

Reply:
That would be the ideal solution. However, for that we would also need to solve the `.*`-type port assignments in the abstract primitives. This is not supported by Yosys IIRC.

Reply:
Have we tried that recently? I'm pretty certain that I've seen the code to support it in the Yosys codebase.

Reply:
I stumbled across this, but I don't remember if I was using an older Yosys version at that point. I did find a couple of sv2v bugs and at some point decided it was best to also upgrade Yosys. So it could very well be that this works now.